app/parse.py

```diff
@@ -8,8 +14,51 @@
     tags: list[str]
 
 
+QUOTE_ROWS = [field.name for field in fields(Quote)]
+
+
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one("span.text").text,
+        author=quote_soup.select_one("small.author").text,
+        tags=[tag.get_text() for tag in quote_soup.select(".tags > a.tag")],
+    )
+
+
+def get_quotes_from_single_page(soup: BeautifulSoup) -> list[Quote]:
+    quotes = []
+
+    products = soup.select(".quote")
+    for product in products:
+        quote = parse_single_quote(product)
+        quotes.append(quote)
+    return quotes
+
+
+def parse_pagination() -> list[Quote]:
+    results = []
+    counter = 1
+    while requests.get(f"{BASE_URL}/page/{counter}/").status_code == 200:
+        print(f"parsed page #{counter}")
+        page = requests.get(f"{BASE_URL}/page/{counter}/").content
+        soup = BeautifulSoup(page, "html.parser")
+        if not get_quotes_from_single_page(soup):
```

Comment:
The condition checks if the list of quotes is empty to break the loop. This logic assumes that an empty page means no more pages are available, which might not always be true. Consider adding a more robust condition or a maximum page limit to avoid infinite loops in case of unexpected HTML structure changes.